includes/class-wc-payments-account.php

```diff
@@ -561,14 +561,23 @@ public function get_cached_account_data() {
 			return [];
 		}
 
-		$account = get_transient( self::ACCOUNT_TRANSIENT );
+		$account_cache = get_option( self::ACCOUNT_OPTION );
+		if ( false === $account_cache || ! isset( $account_cache['account'] ) ) {
+			$account = false;
+		} else {
+			$account = $account_cache['account'];
+
+			// Set $account to false if the cache has expired, triggering another fetch.
+			if ( isset( $account_cache['expires'] ) && $account_cache['expires'] < time() ) {
+				$account = false;
+			}
+		}
```

Comment:
> We could roll it into a private function, something like read_account_from_cache

I think that works well as a name, done in https://github.com/Automattic/woocommerce-payments/pull/1668/commits/e2f4e73a9b8cd464143f633d9bbc70261253e07a.

> In an ideal world, we would add is_valid_cached_account to that private function too. But then there's a problem of telling whether an error has been cached. But maybe we could have a separate private method for it too? We could then store the error bool as a separate key of the array, and make the account to be strongly-typed as a ?array rather than null|array|bool

I'm inclined to keep this change simple for now. There's a lot we can do here, now that we have "properties" (account, expiry) on what we cache I was wondering about making it an object with methods like `is_error`, `has_account` etc. Some enhancements along those lines might let us strongly type a lot of this (and even avoid nullable types).